synapse/logging/context.py

```diff
@@ -720,7 +733,30 @@ def g(*args, **kwargs):
     return g
 
 
-def run_in_background(f, *args, **kwargs) -> defer.Deferred:
+@overload
+def run_in_background(  # type: ignore[misc]
+    f: Callable[..., Awaitable[R]], *args: Any, **kwargs: Any
+) -> "defer.Deferred[R]":
+    # The `type: ignore[misc]` above suppresses
+    # "Overloaded function signatures 1 and 2 overlap with incompatible return types"
```

Comment:
Does mypy still use this overload with the type-ignore?

Reply:
I don't think mypy ever uses the base version of a method when there are `@overload`s. I'm hoping 3-tuples are enough for everybody

Reply:
I guess I just wanted to confirm that this overload is getting used, and that mypy isn't just falling back to `Any` or something due to the type-ignore.

Can we do a reveal_type on something returned by this overload specifically, to be convinced that mypy is using this overload definition?

Reply:
Whoops, my comment above does not make any sense. I thought this was about `gather_results`.

mypy does make use of the overload:
https://mypy-play.net/?mypy=latest&python=3.10&gist=1cbb7ab4a946e9b043218e7b1300694b

Reply:
Brilliant, thank you!